### DISCUSSION TITLE
Cant connect to Modbus RTU to TCP Bridge (Elfin EW11), SDongle works fine

### DISCUSSION THREAD

**Discussion by marvinp93** [wlcrs/huawei_solar#681] [Setting up the integration] [answered]
Hello everyone,

I've been trying to get the integration with an EW11 to work for a few weeks now. Unfortunately, my HA does not get a connection and the integration connection attempt always fails.
I have configured everything as described here ([https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter#serial-connection-to-a-modbus-rtu-to-tcp-bridge](https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter#serial-connection-to-a-modbus-rtu-to-tcp-bridge)). Checked several times. It works via direct connection to the sdongle. I can also see the EW11 in my network.

My system consists of:
Huawei SUN2000-10KTL-M1 with firmware V100R001C00SPC159
SDongleA-05 with firmware V200R022C10SPC108
Luna 10kWh and Power Meter DTSU*** (they shouldn't really matter)

The SDongle and EW11 are connected to my network via a repeater. Since I can access the dongle but not the EW11, I would rule this out as an error for now. Ping to the EW11 also works from the HA.
Does the SDongle possibly have to be pulled out for it to work with the EW11? In that case, however, the WR would not be able to access the WLAN as far as I know. I have also reinstalled the integration in HA and then tried directly with the EW11 only, also no success. I'm at the end of my ideas. Unfortunately I haven't found anyone with the same problem.
Maybe someone else here has an idea?

The connection to the dongle breaks off twice a day and that's exactly why the Wiki advises against this option. That's why I bought the EW11.
Thanks in advance for your help! I'm getting desperate...

**Comment by Roving-Ronin** [comment, marked as answer]
@marvinp93 you can run the sDongle (WiFi or Ethernet to your local network and it allows ONE modbus tcp connection to it) or you can run the Elfin.   Either one, not both. IMHO the sDongle is the preferred option, as it can also connect to FusionSolar portal, allowing reporting back to Huawei for their reporting, remote support and (importantly) to allow pushing firmware updates from FusionSolar down to your inverter/sDongle/LUNA ESS/Optimisers/Modbus etc..
